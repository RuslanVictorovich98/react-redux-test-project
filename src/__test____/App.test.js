import React from 'react';
import {shallow} from 'enzyme';
import ConnectedApp, { App }  from '../App';
import configureStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';


const initialState = {One: 'one'}
const mockStore = configureStore();
const mockFetchData = jest.fn();
let store, wrapper;

const props = {
    fetchUsersRequest: mockFetchData
}


beforeEach(()=>{
    store = mockStore(initialState);
    wrapper = shallow(<App store={store} {...props}/>)
})

describe('TEST component ListCategory', () => {

    it('TEST spapshot to component', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('TEST check call function `fetchUsersRequest()`', () => {
        expect(mockFetchData).toHaveBeenCalled();
    })


})

