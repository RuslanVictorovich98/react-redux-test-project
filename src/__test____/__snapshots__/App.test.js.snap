// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST component ListCategory TEST spapshot to component 1`] = `
<div>
  <Connect(Header) />
  <Container
    fluid={false}
  >
    <ForwardRef
      noGutters={false}
    >
      <Col
        className="left-panel-category"
        lg="3"
        sm="4"
      >
        <Connect(ListCategory) />
      </Col>
      <Col
        lg="9"
        sm="8"
      >
        <Connect(ListProducts) />
      </Col>
    </ForwardRef>
  </Container>
</div>
`;
